readthedocs/core/symlink.py

```diff
@@ -283,3 +284,57 @@ def symlink_versions(self):
             for old_ver in os.listdir(version_dir):
                 if old_ver not in versions:
                     os.unlink(os.path.join(version_dir, old_ver))
+
+
+class PublicSymlink(Symlink):
+    CNAME_ROOT = os.path.join(settings.SITE_ROOT, 'public_cname_root')
+    WEB_ROOT = os.path.join(settings.SITE_ROOT, 'public_web_root')
+    PROJECT_CNAME_ROOT = os.path.join(settings.SITE_ROOT, 'public_cname_project')
+
+    def get_version_queryset(self):
+        return (self.project.versions.protected(only_active=False).filter(built=True) |
+                self.project.versions.protected(only_active=True))
+
+    def get_subprojects(self):
+        return self.project.subprojects.protected()
+
+    def get_translations(self):
+        return self.project.translations.protected()
+
+    def get_default_version(self):
+        default_version = self.project.get_default_version()
+        if self.project.versions.protected().filter(
+            slug=default_version
+        ).exists():
+            return default_version
+        return None
+
+    def run_sanity_check(self):
+        return self.project.privacy_level in [constants.PUBLIC, constants.PROTECTED]
+
+
+class PrivateSymlink(Symlink):
+    CNAME_ROOT = os.path.join(settings.SITE_ROOT, 'private_cname_root')
+    WEB_ROOT = os.path.join(settings.SITE_ROOT, 'private_web_root')
+    PROJECT_CNAME_ROOT = os.path.join(settings.SITE_ROOT, 'private_cname_project')
+
+    def run_sanity_check(self):
+        return self.project.privacy_level in [constants.PRIVATE]
+
+    def get_version_queryset(self):
+        return (self.project.versions.private(only_active=False).filter(built=True) |
+                self.project.versions.private(only_active=True))
+
+    def get_subprojects(self):
+        return self.project.subprojects.private()
+
+    def get_translations(self):
+        return self.project.translations.private()
+
+    def get_default_version(self):
+        default_version = self.project.get_default_version()
+        if self.project.versions.private().filter(
+            slug=default_version
+        ).exists():
```

Comment:
Though there is no stance in pep8 for hanging indents on if statements, I'd rather see this with 2 or 1 levels of indentation, to keep the indentation away from the nested statements:

```python
if self.project.versions.private().filter(
            slug=default_version
        ).exists():
    pass
```

or, though I prefer the above, adding a single level:

```python
if self.project.versions.private().filter(
        slug=default_version
    ).exists():
    pass
```